services/provider/server/server.go

```diff
@@ -140,6 +140,28 @@ func (s *OCSProviderServer) OnboardConsumer(ctx context.Context, req *pb.Onboard
 		klog.Errorf("failed to validate onboarding ticket for consumer %q. %v", req.ConsumerName, err)
 		return nil, status.Errorf(codes.InvalidArgument, "onboarding ticket is not valid. %v", err)
 	}
+
+	// get the consumer corresponding to this ticket
+	storageConsumerList := &ocsv1alpha1.StorageConsumerList{}
+	if err := s.client.List(ctx, storageConsumerList, client.InNamespace(s.namespace)); err != nil {
+		klog.Errorf("failed to get storageconsumers in the namespace: %v", err)
+		return nil, status.Errorf(codes.Internal, "failed to get storageconsumers. %v", err)
+	} else if len(storageConsumerList.Items) < 1 {
+		return nil, status.Errorf(codes.FailedPrecondition, "no storageconsumers exist in the namespace")
+	}
+
+	consumerName := ""
+	for idx := range storageConsumerList.Items {
+		consumer := &storageConsumerList.Items[idx]
+		if consumer.GetAnnotations()[TicketAnnotation] == onboardingTicket.ID {
+			consumerName = consumer.Name
+			break
+		}
+	}
+	if consumerName == "" {
+		return nil, status.Errorf(codes.FailedPrecondition, "no storageConsumer found for supplied onboarding ticket")
+	}
```

Comment:
If you have the ticket, you shouldn't need to list consumers or secrets.

Reply:
ack, done.